src/async.rs

```diff
@@ -155,6 +155,14 @@ impl<'a, T> SendFut<'a, T> {
         }
     }
 
+    /// Consume `self` returning the queued value if it has not yet been sent.
+    pub fn into_inner(mut self) -> Option<T> {
```

Comment:
Could this be `&mut self`, or is it intentionally consuming the SendFut?

Reply:
I think it can be made to accept `&mut self`, yes.